network/http/detectors-http.tf

```diff
@@ -93,47 +93,47 @@ resource "signalfx_detector" "http_content_length" {
     parameterized_subject = local.rule_subject
     parameterized_body    = local.rule_body
   }
 }
 
 resource "signalfx_detector" "certificate_expiration_date" {
-  name = format("%s %s", local.detector_name_prefix, "TLS certificate expiring in ")
+  name = format("%s %s", local.detector_name_prefix, "TLS certificate expiring")
 
   program_text = <<-EOF
-    A = data('http.cert_expiry', filter=${module.filter-tags.filter_custom})${var.certificate_expiration_date_aggregation_function}${var.certificate_expiration_date_transformation_function}
+    A = data('http.cert_expiry', filter=${module.filter-tags.filter_custom}, rollup='min')${var.certificate_expiration_date_aggregation_function}${var.certificate_expiration_date_transformation_function}
     signal = (A/86400).publish('signal')
-    detect(when(signal < ${var.certificate_expiration_date_threshold_critical})).publish('CRIT')
-    detect(when(signal < ${var.certificate_expiration_date_threshold_major}) and when(signal >= ${var.certificate_expiration_date_threshold_critical})).publish('MAJOR')
+    detect(when(signal < ${var.certificate_expiration_date_threshold_major})).publish('MAJOR')
+    detect(when(signal < ${var.certificate_expiration_date_threshold_minor}) and when(signal >= ${var.certificate_expiration_date_threshold_major})).publish('MINOR')
 EOF
 
   rule {
-    description           = " < ${var.certificate_expiration_date_threshold_critical} days"
-    severity              = "Critical"
-    detect_label          = "CRIT"
-    disabled              = coalesce(var.certificate_expiration_date_disabled_critical, var.certificate_expiration_date_disabled, var.detectors_disabled)
-    notifications         = coalescelist(lookup(var.certificate_expiration_date_notifications, "critical", []), var.notifications.critical)
+    description           = "is too low < ${var.certificate_expiration_date_threshold_major} days"
+    severity              = "Major"
+    detect_label          = "MAJOR"
+    disabled              = coalesce(var.certificate_expiration_date_disabled_major, var.certificate_expiration_date_disabled, var.detectors_disabled)
+    notifications         = coalescelist(lookup(var.certificate_expiration_date_notifications, "major", []), var.notifications.major)
     parameterized_subject = local.rule_subject
     parameterized_body    = local.rule_body
   }
 
   rule {
-    description           = " < ${var.certificate_expiration_date_threshold_major} days"
-    severity              = "Major"
-    detect_label          = "MAJOR"
-    disabled              = coalesce(var.certificate_expiration_date_disabled_major, var.certificate_expiration_date_disabled, var.detectors_disabled)
-    notifications         = coalescelist(lookup(var.certificate_expiration_date_notifications, "major", []), var.notifications.major)
+    description           = "is too low < ${var.certificate_expiration_date_threshold_minor} days"
```

Comment:
`     description           = "expires very soon: ${var.certificate_expiration_date_threshold_minor} days"`

Reply:
If I do that I will have to remove "expiring" from the name and it will be equal to the invalid detector one. Name are not used as ID and this is technically possible but I prefer to keep every name unique to list them in readme for example.

I propose to set : 

- the name as "TLS certificate expiry date"
- rule as "is too soon < X days"

what do you think?

Reply:
Seems good to me